sources/en/r/ranobes.py

```diff
@@ -1,109 +1,104 @@
 # -*- coding: utf-8 -*-
 import logging
+import re
 
-import js2py
-from bs4.element import Tag
+from typing import Generator, Union
 
-from lncrawl.core.crawler import Crawler
+from bs4 import BeautifulSoup, Tag
+
+from lncrawl.models import Chapter, SearchResult, Volume
+from lncrawl.templates.browser.searchable import SearchableBrowserTemplate
+from lncrawl.core.exeptions import FallbackToBrowser
+
+from urllib.parse import urljoin, quote_plus
 
 logger = logging.getLogger(__name__)
 
 
-class RanobeLibCrawler(Crawler):
+digit_regex = re.compile(r"\/(\d+)-")
+
+
+class RanobeLibCrawler(SearchableBrowserTemplate):
     base_url = [
-        "http://ranobes.net/",
-        "https://ranobes.net/",
+        "https://ranobes.top/",
     ]
```

Comment:
Might be more sensible to leave `ranobes.net` in and include an `http` link for `.top`.

It wouldn't bring any trouble since ranobes.net redirects to ranobes.top, and it could be implemented with a simple `self.absolute_url`, or something similar.

If we don't consider backwards compatibility, people still using `.net` links, without the appropriate knowledge, would be thoroughly confused, for good reason. Or might even believe that `lncrawler` doesn't support `ranobes`, which is an erroneous, but easy conclusion to make.

Reply:
Done in https://github.com/dipu-bd/lightnovel-crawler/pull/1990/commits/68f7c130111092eadd3ac6745a9f1ce8a8ad4685